### PR TITLE
[CP-1484] Add new mechanism for data synchronization between MOS and MC

### PR DESCRIPTION
**Pull request by pawel-mudita**
This mechanism is similar to the current backup implementation but is stripped of irrelevant data that is not required by the MC to perform data synchronization.
Documentation: https://appnroll.atlassian.net/wiki/spaces/CP/pages/1727627287/Backup+Endpoint+API+4#Synchronization
**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [x] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

**Review by swichu91 — APPROVED** [review]

File: module-services/service-db/DBServiceAPI.cpp [hunk 1]
```diff
@@ -275,6 +275,21 @@ auto DBServiceAPI::DBBackup(sys::Service *serv, std::string backupPath) -> bool
     return false;
 }
 
+auto DBServiceAPI::DBPrepareSyncPackage(sys::Service *serv, const std::string &syncPackagePath) -> bool
+{
+    LOG_INFO("DBPrepareSyncPackage %s", syncPackagePath.c_str());
+
+    std::shared_ptr<DBServiceMessageSyncPackage> msg =
```
Comment (SP2FET) [inline module-services/service-db/DBServiceAPI.cpp:282]:
```suggestion
    auto msg =
```

File: module-services/service-db/DBServiceAPI.cpp [hunk 2]
```diff
@@ -275,6 +275,21 @@ auto DBServiceAPI::DBBackup(sys::Service *serv, std::string backupPath) -> bool
...
+        std::make_shared<DBServiceMessageSyncPackage>(MessageType::DBSyncPackage, syncPackagePath);
+
+    auto ret = serv->bus.sendUnicastSync(msg, service::name::db, DefaultTimeoutInMs);
```
Comment (SP2FET) [inline module-services/service-db/DBServiceAPI.cpp:285]:
```suggestion
    auto ret = serv->bus.sendUnicastSync(std::move(msg), service::name::db, DefaultTimeoutInMs);
```

File: module-services/service-db/include/service-db/DBServiceMessage.hpp [hunk 3]
```diff
@@ -17,6 +17,13 @@ class DBServiceMessageBackup : public DBMessage
     std::string backupPath;
 };
 
+class DBServiceMessageSyncPackage : public DBMessage
+{
+  public:
+    DBServiceMessageSyncPackage(MessageType messageType, std::string syncPackagePath);
+    std::string syncPackagePath;
```
Comment (SP2FET) [inline module-services/service-db/include/service-db/DBServiceMessage.hpp:24]: this field has to be public?

File: module-services/service-db/messages/DBServiceMessage.cpp [hunk 4]
```diff
@@ -7,5 +7,9 @@ DBServiceMessageBackup::DBServiceMessageBackup(MessageType messageType, std::str
     : DBMessage(messageType), backupPath(backupPath)
 {}
 
+DBServiceMessageSyncPackage::DBServiceMessageSyncPackage(MessageType messageType, std::string syncPackagePath)
```
Comment (SP2FET) [inline module-services/service-db/messages/DBServiceMessage.cpp:10]: if the path won't be modified consider passing const-ref ;)

File: module-services/service-desktop/ServiceDesktop.cpp [hunk 5]
```diff
@@ -101,20 +102,38 @@ std::string ServiceDesktop::prepareBackupFilename()
     return std::string(backupFileName.data());
 }
 
+std::string ServiceDesktop::prepareSyncFilename()
+{
+    std::array<char, 64> syncFileName{};
```
Comment (SP2FET) [inline module-services/service-desktop/ServiceDesktop.cpp:107]: magic number?

File: module-services/service-desktop/endpoints/backup/BackupHelper.cpp [hunk 6]
```diff
@@ -59,15 +65,75 @@ namespace sdesktop::endpoints
...
+                context.setResponseBody(ownerServicePtr->getBackupSyncRestoreStatus());
+            }
+            else {
+                context.setResponseStatus(http::Code::NotFound);
+            }
+        }
+        else {
+            LOG_DEBUG("Backup task not found");
+            context.setResponseStatus(http::Code::BadRequest);
+        }
```
Comment (SP2FET) [inline module-services/service-desktop/endpoints/backup/BackupHelper.cpp:87]: those if's don't look nice :( but I assume it's the only way to write it down?

File: module-vfs/paths/filesystem_paths.cpp [hunk 7]
```diff
@@ -70,6 +71,11 @@ namespace purefs
             return std::filesystem::path{eMMC_disk} / PATH_USER / PATH_BACKUP;
         }
 
+        std::filesystem::path getSyncPackagePath() noexcept
+        {
+            return std::filesystem::path{eMMC_disk} / PATH_USER / PATH_SYNC;
```
Comment (SP2FET) [inline module-vfs/paths/filesystem_paths.cpp:76]:
```suggestion
            return getUserDiskPath() / PATH_SYNC;
```

File: module-services/service-db/include/service-db/DBServiceMessage.hpp (hunk 3 above)
Comment (pawel-mudita) [reply module-services/service-db/include/service-db/DBServiceMessage.hpp:24]: Yes, its needed here: `/products/PurePhone/services/db/ServiceDB.cpp:211`
However, I'll check if I can write it differently during https://appnroll.atlassian.net/browse/CP-1427

File: module-services/service-db/DBServiceAPI.cpp (hunk 1 above)
Comment (pawel-mudita) [reply module-services/service-db/DBServiceAPI.cpp]: done

File: module-services/service-db/DBServiceAPI.cpp (hunk 2 above)
Comment (pawel-mudita) [reply module-services/service-db/DBServiceAPI.cpp]: done

File: module-services/service-db/messages/DBServiceMessage.cpp (hunk 4 above)
Comment (pawel-mudita) [reply module-services/service-db/messages/DBServiceMessage.cpp]: done

File: module-services/service-desktop/ServiceDesktop.cpp (hunk 5 above)
Comment (pawel-mudita) [reply module-services/service-desktop/ServiceDesktop.cpp]: changed to a const variable with meaningful name

File: module-vfs/paths/filesystem_paths.cpp (hunk 7 above)
Comment (pawel-mudita) [reply module-vfs/paths/filesystem_paths.cpp]: done

**Review by SP2FET — APPROVED** [review]

**Pull request merged**